skbio/stats/composition.py

```diff
@@ -807,6 +807,319 @@ def centralize(mat):
+    # Set up input and parameters
+    kwargs = { 
+        "x":x,
+        "y":y,
+        "ddof":ddof,
+    }
+
+    # Run backend function 
+    if robust:
+        return _robust_vlr(**kwargs) 
+    else:
+        return _vlr(**kwargs)
+    
+
+@experimental(as_of="0.5.7")
+def _pairwise_vlr(mat:np.array, ddof:int):
+    r"""
+    Performs pairwise variance log ratio transformation
+    
+    Parameters
+    ----------
+    mat : array_like, float
+       a matrix of proportions where
+       rows = compositions and
+       columns = components
+       
+    ids: array_like, str
+        component names
+        
+    ddof: int
+        degrees of freedom
+       
+    Returns
+    -------
+    skbio.DistanceMatrix
+         distance matrix of variance log ratio values
+         
+    
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S, Bähler J (2015) 
+           Proportionality: A Valid Alternative to Correlation for Relative Data. 
+           PLoS Comput Biol 11(3): e1004075. https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C. 
+           How should we measure proportionality on relative gene expression data?. 
+           Theory Biosci. 135, 21–36 (2016). https://doi.org/10.1007/s12064-015-0220-8
+    """
+        
+    # Log Transform
+    X_log = np.log(mat)
+    
+    # Variance Log Ratio
+    covariance = np.cov(X_log.T, ddof=ddof) 
+    diagonal = np.diagonal(covariance)
+    vlr_data = -2*covariance + diagonal[:,np.newaxis] + diagonal 
+    return vlr_data
+
+@experimental(as_of="0.5.7")
+def _robust_pairwise_vlr(mat:np.array, ddof:int):
+    r"""
+    Performs pairwise variance log ratio transformation while masking zeros
+    
+    Parameters
+    ----------
+    mat : array_like, float
+       a matrix of proportions where
+       rows = compositions and
+       columns = components
+       
+    ids: array_like, str
+        component names
+        
+    ddof: int
+        degrees of freedom
+       
+    Returns
+    -------
+    skbio.DistanceMatrix
+         distance matrix of variance log ratio values
+         
+    
+    References
+    ----------
+    .. [1] V. Lovell D, Pawlowsky-Glahn V, Egozcue JJ, Marguerat S, Bähler J (2015) 
+           Proportionality: A Valid Alternative to Correlation for Relative Data. 
+           PLoS Comput Biol 11(3): e1004075. https://doi.org/10.1371/journal.pcbi.1004075
+    .. [2] Erb, I., Notredame, C. 
+           How should we measure proportionality on relative gene expression data?. 
+           Theory Biosci. 135, 21–36 (2016). https://doi.org/10.1007/s12064-015-0220-8
+    """
+    # Mask zeros
+    X = np.ma.masked_array(mat, mask=mat == 0)
+    
+    # Log Transform
+    X_log = np.ma.log(X)
+    
+    # Variance Log Ratio
+    covariance = np.ma.cov(X_log.T, ddof=ddof) 
+    diagonal = np.ma.diagonal(covariance)
+    vlr_data = -2*covariance + diagonal[:,np.newaxis] + diagonal 
+    return vlr_data
```

Comment:
Finally getting around to playing with this.  Turns out that vlr_data is a masked array, and the pairs that don't have any common samples will be filled with some random number (i.e. -2).  This won't play nicely with the distance matrix object since you can't have negative distances.

There are a couple of ways we can deal with this.
1. Have this as a separate method (i.e. `masked_pairwise_vlr`) where we just return the masked array and let the user deal with it.  We won't have DistanceMatrix compatibility anymore, but what gives ...
2. Have a parameter in `_robust_vlr` called `fill_value` set to an arbitrary high constant (i.e. 1e6) that fills these missing values to make the DistanceMatrix methods happy.

Reply:
@mortonjt this is interesting.  Can you show an example so I can figure out how we can implement this? I just submitted my PR.  I've almost fixed all of the formatting requests to pass tests.  I could use a little help with these ones: 
```
skbio/stats/composition.py:1063:5: E128 continuation line under-indented for visual indent
skbio/stats/composition.py:1064:5: E128 continuation line under-indented for visual indent
skbio/stats/composition.py:1065:5: E128 continuation line under-indented for visual indent
skbio/stats/composition.py:1066:5: E128 continuation line under-indented for visual indent
skbio/stats/composition.py:1066:5: E125 continuation line with same indent as next logical line
```

Reply:
You can pull out any microbiome dataset and probably see this issue.  I used the [Global Gut dataset](https://qiita.ucsd.edu/study/description/850) and only looked the top 100 microbes.

I was also noticing that there were some negative VLR values, which was weird.